addons/mrp/models/stock_move.py

```diff
@@ -486,3 +486,8 @@ def _multi_line_quantity_done_set(self, quantity_done):
             self.move_line_ids = self._set_quantity_done_prepare_vals(quantity_done)
         else:
             super()._multi_line_quantity_done_set(quantity_done)
+
+    def _prepare_procurement_values(self):
+        res = super()._prepare_procurement_values()
+        res['bom_line_id'] = self.bom_line_id.id
```

Comment:
other stock.move()._prepare_procurement_values method all seem to use recordset as value instead of IDs, but if down the line we are doing create/write I guess it should be the same